core/templates/services/rte-helper-modal.controller.ts

```diff
@@ -62,18 +82,57 @@ angular.module('oppia').controller('RteHelperModalController', [
       $scope.$broadcast('externalSave');
 
       var customizationArgsDict = {};
-      for (var i = 0; i < $scope.tmpCustomizationArgs.length; i++) {
-        var caName = $scope.tmpCustomizationArgs[i].name;
-        if (caName === 'video_id') {
-          var temp = $scope.tmpCustomizationArgs[i].value;
-          customizationArgsDict[caName] = (
-            extractVideoIdFromVideoUrl(temp.toString()));
-        } else {
-          customizationArgsDict[caName] = (
-            $scope.tmpCustomizationArgs[i].value);
+      let caName : string;
+      if ($scope.currentRteIsMathExpressionEditor) {
+        var svgFile = $scope.tmpCustomizationArgs[0].value.svgFile;
+        var svgFileName = $scope.tmpCustomizationArgs[0].value.svg_filename;
+        if (
+          ContextService.getImageSaveDestination() ===
+          IMAGE_SAVE_DESTINATION_LOCAL_STORAGE) {
+          ImageLocalStorageService.saveImage(svgFileName, svgFile);
+          var mathContentDict = {
+            raw_latex: $scope.tmpCustomizationArgs[0].value.raw_latex,
+            svg_filename: svgFileName
+          };
+          caName = 'math_content';
+          customizationArgsDict[caName] = mathContentDict;
+          $uibModalInstance.close(customizationArgsDict);
+          $scope.currentRteIsMathExpressionEditor = false;
+          return;
+        }
+        var resampledFile = (
+          ImageUploadHelperService.convertImageDataToImageFile(svgFile));
+        AssetsBackendApiService.saveMathExpresionImage(
+          resampledFile, svgFileName, ContextService.getEntityType(),
+          ContextService.getEntityId()).then(function(response) {
+          var mathContentDict = {
+            raw_latex: $scope.tmpCustomizationArgs[0].value.raw_latex,
+            svg_filename: response.filename
+          };
+          caName = 'math_content';
```

Comment:
WHy do we need to define this? will it be same as $scope.tmpCustomizationArgs[i].name?

Reply:
done, changed it